tsl/src/bgw_policy/compression_api.c

```diff
@@ -257,24 +254,30 @@ policy_compression_remove(PG_FUNCTION_ARGS)
 {
 	Oid hypertable_oid = PG_GETARG_OID(0);
 	bool if_exists = PG_GETARG_BOOL(1);
+	Hypertable *ht;
+	Cache *hcache;
 
 	PreventCommandIfReadOnly("remove_compression_policy()");
 
-	int ht_id = ts_hypertable_relid_to_id(hypertable_oid);
+	ht = ts_hypertable_cache_get_cache_and_entry(hypertable_oid, CACHE_FLAG_NONE, &hcache);
 
 	List *jobs = ts_bgw_job_find_by_proc_and_hypertable_id(POLICY_COMPRESSION_PROC_NAME,
 														   INTERNAL_SCHEMA_NAME,
-														   ht_id);
+														   ht->fd.id);
+
+	ts_cache_release(hcache);
```

Comment:
I am not sure why you rewrite this code. `ts_hypertable_relid_to_id` does exactly this and the hypertable structure is not used later. You only increase the risk of mistakes when maintaining this code.

Reply:
It feels strange to have a function that returns `InvalidOid` so I would expect that `ts_hypertable_relid_to_id` to be fixed instead (and generate an error). However, that is a minor issue. I agree that we should not have several different locations to generate similar errors from since that leads to inconsistent error messages.

Reply:
I agree, but that's a separate refactoring, IMO.